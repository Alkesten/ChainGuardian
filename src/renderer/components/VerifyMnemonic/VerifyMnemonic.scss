.verify-mnemonic-container{
    width: 640px;
}
.verify-button-container{
    margin-top: 64px;
    display: flex;
    justify-content: space-evenly;
}