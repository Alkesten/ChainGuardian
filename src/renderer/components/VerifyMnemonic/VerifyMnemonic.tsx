import * as React from "react";
import {ButtonInverted, ButtonDestructive, ButtonPrimary} from "../Button/ButtonStandard";
import {useState} from "react";

export interface IVerifyMnemonicProps {
    question: string;
    answers: string[];
    correctAnswer: string;
    onCorrectAnswer:  () => void;
    onInvalidAnswer: () => void;
}

export const VerifyMnemonic: React.FunctionComponent<IVerifyMnemonicProps> = (
    props: IVerifyMnemonicProps) => {
    const [clicked, setClicked] = useState();
    function isCorrect (
        answer: string, 
        correctAnswer: string, 
        onCorrectAnswer: () => void,
        onInvalidAnswer: () => void): void {
        answer === correctAnswer ? onCorrectAnswer() : onInvalidAnswer();
        setClicked(answer);
    }

    return(
        <div className="verify-mnemonic-container">
            <h1>{props.question}</h1>
            <div className="verify-button-container">
                {props.answers.map(answer => {
                    if(clicked===answer && (answer === props.correctAnswer)){
                        return (
                            <ButtonPrimary key={answer} onClick={(): void => isCorrect(
                                answer, 
                                props.correctAnswer, 
                                props.onCorrectAnswer, 
                                props.onInvalidAnswer)}>{answer}</ButtonPrimary>);
                    } else if(clicked===answer && (answer !== props.correctAnswer)){
                        return (
                            <ButtonDestructive key={answer} onClick={(): void => isCorrect(
                                answer, 
                                props.correctAnswer, 
                                props.onCorrectAnswer, 
                                props.onInvalidAnswer)}>{answer}</ButtonDestructive>);
                    } else{
                        return (
                            <ButtonInverted key={answer} onClick={(): void => isCorrect(
                                answer, 
                                props.correctAnswer, 
                                props.onCorrectAnswer, 
                                props.onInvalidAnswer)}>{answer}</ButtonInverted>);
                    }
                })}
            </div>
        </div>
    );
};


