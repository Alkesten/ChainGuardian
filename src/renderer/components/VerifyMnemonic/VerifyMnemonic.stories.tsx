import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select, object, array } from '@storybook/addon-knobs';
import { VerifyMnemonic } from './VerifyMnemonic';

storiesOf('Verify Mnemonic', module).add('Verify Mnemonic', () => {
    const question = text("question", "What's the 4th word in the mnemonic?");
    const answers = array("answers",["Accept","Call","Iguanas"]);
    const correctAnswer = text("correct answer", "Iguanas")
    return <div >
            <VerifyMnemonic 
                question={question}
                answers={answers}
                correctAnswer={correctAnswer}
                onCorrectAnswer={(): void => console.log("correct")}
                onInvalidAnswer={(): void => console.log("invalid")}/>
        </div>;
}).addDecorator(withKnobs);