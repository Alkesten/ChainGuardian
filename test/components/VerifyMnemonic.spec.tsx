import * as React from "react";
import * as renderer from "react-test-renderer";
import {VerifyMnemonic} from "../../src/renderer/components/VerifyMnemonic/VerifyMnemonic";

describe("Verify Mnemonic", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<VerifyMnemonic 
                question="Test question?"
                answers={["Accept","Call","Iguanas"]}
                correctAnswer="Iguanas"
                onCorrectAnswer={(): void => console.log("correct")}
                onInvalidAnswer={(): void => console.log("invalid")}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});