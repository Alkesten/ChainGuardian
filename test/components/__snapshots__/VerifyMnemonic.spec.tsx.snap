// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Mnemonic renders correctly 1`] = `
<div
  className="verify-mnemonic-container"
>
  <h1>
    Test question?
  </h1>
  <div
    className="verify-button-container"
  >
    <button
      className="button btn-inverted "
      datafield="false"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className=""
      >
        Accept
      </div>
    </button>
    <button
      className="button btn-inverted "
      datafield="false"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className=""
      >
        Call
      </div>
    </button>
    <button
      className="button btn-inverted "
      datafield="true"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className=""
      >
        Iguanas
      </div>
    </button>
  </div>
</div>
`;
